fw/vhost.c

```diff
@@ -1345,8 +1347,16 @@ tfw_location_new(TfwVhost *vhost, tfw_match_t op, const char *arg, size_t len)
 		return NULL;
 	vhost->loc_sz++;
 
-	if (tfw_frang_cfg_inherit(loc->frang_cfg, vhost->loc_dflt->frang_cfg))
-		return NULL;
+	if (strncasecmp(vhost->name.data, TFW_VH_DFT_NAME, vhost->name.len)) {
+		if (tfw_frang_cfg_inherit(loc->frang_cfg,
```

Comment:
Please look at this function `tfw_frang_cfg_inherit` we have BUG in it.
if (!curr->http_ct_vals)
 we only set r = -ENOMEM and do not return. Then we memcpy(curr->http_ct_vals, from->http_ct_vals, sz);

Reply:
Add else to avoid using curr->http_ct_vals after error.